src/fireedge/src/server/routes/api/auth/functions.js

```diff
@@ -14,566 +14,188 @@
-  const iv = getIV()
-  if (serverAdmin && username && key && iv) {
-    const expire = parseInt(expireTime.toSeconds(), 10)
-    rtn = {
-      token: encrypt(
-        `${serverAdmin}:${username}:${expire}`,
-        key,
-        iv
-      ),
-      time: expire
-    }
+      fillHookResource: false,
+    })
+  } else {
+    next()
   }
-  return rtn
 }
 
 /**
- * Wrap user with serveradmin.
- *
- * @param {object} serverAdminData - opennebula serveradmin data
- * @param {object} userData - opennebula user data
- */
-const wrapUserWithServerAdmin = (serverAdminData = {}, userData = {}) => {
-  const relativeTime = getRelativeTime()
-  let serverAdminName = ''
-  let serverAdminPassword = ''
-  let userName = ''
-
-  if (
-    relativeTime &&
-    serverAdminData &&
-    serverAdminData.USER &&
-    (serverAdminName = serverAdminData.USER.NAME) &&
-    (serverAdminPassword = serverAdminData.USER.PASSWORD) &&
-    userData &&
-    (userName = userData.NAME) &&
-    userData.ID &&
-    userData.TEMPLATE
-  ) {
-    /*********************************************************
-      * equals what is placed in:
-      * src/authm_mad/remotes/server_cipher/server_cipher_auth.rb:44
-    *********************************************************/
-    setKey(serverAdminPassword.substring(0, 32))
-    setIV(serverAdminPassword.substring(0, 16))
-
-    const JWTusername = `${serverAdminName}:${userName}`
-
-    let tokenWithServerAdmin
-    let setGlobalNewToken = false
-    const validToken = getCreatedTokenOpennebula(JWTusername)
-    if (validToken) {
-      tokenWithServerAdmin = validToken
-    } else {
-      setGlobalNewToken = true
-      tokenWithServerAdmin = createTokenServerAdmin(serverAdminName, userName)
-    }
-
-    if (tokenWithServerAdmin) {
-      genJWT(
-        tokenWithServerAdmin,
-        {
-          NAME: JWTusername,
-          ID: userData.ID,
-          TEMPLATE: userData.TEMPLATE
-        }
-      )
-
-      // set global state
-      if (setGlobalNewToken) {
-        if (!global.users) {
-          global.users = {}
-        }
-        if (!global.users[JWTusername]) {
-          global.users[JWTusername] = { tokens: [] }
-        }
-        global.users[JWTusername].tokens.push({ token: tokenWithServerAdmin.token, time: parseInt(expireTime.toSeconds(), 10) })
-      }
-      next()
+ * Fireedge user remote auth.
+ *
+ * @param {object} res - http response
+ * @param {Function} next - express stepper
+ * @param {object} params - params of http request
+ * @param {object} _ - user of http request
+ * @param {Function} oneConnection - function of xmlrpc
+ * @param {string} typeAuth - auth type
+ */
+const remoteAuth = (
+  res = {},
+  next = defaultEmptyFunction,
+  params = {},
+  _,
+  oneConnection = defaultEmptyFunction,
+  typeAuth
+) => {
+  const { user } = params
+  setNext(next)
+  setNodeConnect(oneConnection)
+  updaterResponse(new Map(internalServerError).toObject())
+  if (user) {
+    switch (typeAuth) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are using less than 3 statements in `switch`. It&#39;s better to use `if-else` instead.